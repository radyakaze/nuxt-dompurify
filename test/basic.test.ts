import { fileURLToPath } from 'node:url'
import { describe, it, expect } from 'vitest'
import { setup, $fetch } from '@nuxt/test-utils/e2e'
import { JSDOM } from 'jsdom'

describe('ssr', async () => {
  await setup({
    rootDir: fileURLToPath(new URL('./fixtures/profiles', import.meta.url)),
  })

  it('should purify html', async () => {
    // Get response to a server-rendered page with `$fetch`.
    const html: string = await $fetch('/')

    const dom = new JSDOM(html)

    expect(dom.window.document.getElementById('test')?.innerHTML).toMatchSnapshot()
  })

  it('should purify html with heading profile that does not exist', async () => {
    // Get response to a server-rendered page with `$fetch`.
    const html: string = await $fetch('/heading')

    const dom = new JSDOM(html)

    expect(dom.window.document.getElementById('test')?.innerHTML).toMatchSnapshot()
  })
})
