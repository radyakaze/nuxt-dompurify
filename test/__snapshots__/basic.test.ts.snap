// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ssr > should purify html 1`] = `
"
<div>
  <h1>Welcome to My Website</h1>
  <h3>This is H3 heading</h3>
  <p>This is a <strong>simple</strong> paragraph.</p>
  <img src="image.jpg">
  <a href="https://example.com">Click me!</a>
  
</div>"
`;

exports[`ssr > should purify html with heading profile that does not exist 1`] = `
"

  <h1>Welcome to My Website</h1>
  <h3>This is H3 heading</h3>
  This is a simple paragraph.
  
  Click me!
  
"
`;
